RecoParticleFlow/PFRecHitProducer/plugins/LegacyPFRecHitProducer.cc

```diff
@@ -54,23 +54,28 @@ void LegacyPFRecHitProducer::produce(edm::Event& event, const edm::EventSetup& s
   const reco::PFRecHitHostCollection::ConstView& alpakaPfRecHits = pfRecHitsAlpakaSoA.const_view();
 
   reco::PFRecHitCollection out;
-  out.reserve(alpakaPfRecHits.size());
+  int reserveSize = 0;
+  if (alpakaPfRecHits.metadata().size() != 0)
+    reserveSize = alpakaPfRecHits.size();
+  out.reserve(reserveSize);
 
-  for (size_t i = 0; i < alpakaPfRecHits.size(); i++) {
-    reco::PFRecHit& pfrh =
-        out.emplace_back(caloGeo_.at(alpakaPfRecHits[i].layer())->getGeometry(alpakaPfRecHits[i].detId()),
-                         alpakaPfRecHits[i].detId(),
-                         alpakaPfRecHits[i].layer(),
-                         alpakaPfRecHits[i].energy());
-    pfrh.setTime(alpakaPfRecHits[i].time());
-    pfrh.setDepth(alpakaPfRecHits[i].depth());
+  if (alpakaPfRecHits.metadata().size() != 0) {
```

Comment:
this could be more simply
```c++
reco::PFRecHitCollection out;

if (alpakaPfRecHits.metadata().size() != 0) {
  out.reserve(alpakaPfRecHits.size());
  for (size_t i = 0; i < alpakaPfRecHits.size(); i++) {
    ...
```